### PR TITLE
feat(obstacle_avoidance_planner): add options for plan_from_ego

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

- add `plan_from_ego` and `max_plan_from_ego_length` params
   - disabled defaultly
- plan_from_ego is enabled when
  - trajectory length is shorter than `max_plan_from_ego_length`
  - stopped
- remove adjustment of error_weight_ratio  when plan_from_ego
  - `constexpr double obstacle_avoid_error_weight_ratio = 1 / 100.0;` 
## Related links

launch : https://github.com/tier4/autoware_launch/pull/371

## Tests performed

Checked that plan_from_ego is enabled when following conditions are true with psim
  - trajectory length is shorter than `max_plan_from_ego_length`
  - stopped

(defaultly disabled but if enabled)
These are TIER IV internal test results but I share them for your information.
both branches are based on v0.31
- disabled
  - https://evaluation.tier4.jp/evaluation/reports/f5f855d1-5383-5c53-b3e3-f8ac5d6f225b?project_id=prd_jt
  - 260/292 passed
- enabled
  - https://evaluation.tier4.jp/evaluation/reports/dbebb836-d5f4-597c-887c-ac155495afac?project_id=prd_jt
  - 262/292 passed

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kosuke55** [comment]
send PR for launch https://github.com/tier4/autoware_launch/pull/371

File: planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp [hunk 1]
```diff
@@ -411,8 +411,18 @@ void MPTOptimizer::calcPlanningFromEgo(std::vector<ReferencePoint> & ref_points)
...
-  const bool plan_from_ego =
-    mpt_param_.plan_from_ego && std::abs(current_ego_vel_) < epsilon && ref_points.size() > 1;
+  const double trajectory_length = [&]() {
+    double sum_length = 0;
+    for (size_t i = 1; i < ref_points.size(); ++i) {
+      sum_length +=
+        tier4_autoware_utils::calcDistance2d(ref_points.at(i - 1).p, ref_points.at(i).p);
+    }
+    return sum_length;
+  }();
```
Comment (takayuki5168) [inline planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp:421]: can be replaced with tier4_autoware_utils::calcArcLength();
If ReferencePoint msg is not supported in this function, ignore this comment.

File: planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp [hunk 2]
```diff
@@ -633,25 +643,8 @@ MPTOptimizer::ValueMatrix MPTOptimizer::generateValueMatrix(
...
-    const bool near_kinematic_state_while_planning_from_ego = [&]() {
-      const size_t min_idx = static_cast<size_t>(std::max(0, static_cast<int>(i) - 20));
-      const size_t max_idx = std::min(ref_points.size() - 1, i + 20);
-      for (size_t j = min_idx; j <= max_idx; ++j) {
-        if (ref_points.at(j).plan_from_ego && ref_points.at(j).fix_kinematic_state) {
-          return true;
-        }
-      }
-      return false;
-    }();
```
Comment (takayuki5168) [inline planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp:646]: Do not remove this please, should be commented out instead.

File: planning/obstacle_avoidance_planner/include/obstacle_avoidance_planner/mpt_optimizer.hpp [hunk 3]
```diff
@@ -128,6 +128,7 @@ struct ReferencePoint
   //       second is fixing current ego pose when no velocity for planning from ego pose
   boost::optional<Eigen::Vector2d> fix_kinematic_state = boost::none;
   bool plan_from_ego = false;
+  double max_plan_from_ego_length = 10.0;
```
Comment (takayuki5168) [inline planning/obstacle_avoidance_planner/include/obstacle_avoidance_planner/mpt_optimizer.hpp:131]: This is not required in ReferencePoint.

File: planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp (hunk 1 above)
Comment (kosuke55) [reply planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp]: can use calcArcLength for ReferencePoint so replace it.
https://github.com/autowarefoundation/autoware.universe/pull/1191/commits/3147a55e5a2c447e15db7d0d8be9727cc9fdc47d

File: planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp (hunk 2 above)
Comment (kosuke55) [reply planning/obstacle_avoidance_planner/src/mpt_optimizer.cpp]: add it again as coment. https://github.com/autowarefoundation/autoware.universe/pull/1191/commits/7b8fb4febd4e73938894e2b0c494d6715ded9e68

File: planning/obstacle_avoidance_planner/include/obstacle_avoidance_planner/mpt_optimizer.hpp (hunk 3 above)
Comment (kosuke55) [reply planning/obstacle_avoidance_planner/include/obstacle_avoidance_planner/mpt_optimizer.hpp]: Remove it https://github.com/autowarefoundation/autoware.universe/pull/1191/commits/8a03c11cfc83d9870d6ec8f3388ffa9fee5890e8

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Pull request merged**